python/catkin/builder.py

```diff
@@ -922,9 +923,15 @@ def build_workspace_isolated(
                 _print_build_error(package, e)
                 # Let users know how to reproduce
                 # First add the cd to the build folder of the package
-                cmd = 'cd ' + os.path.join(buildspace, package.name) + ' && '
+                cmd = 'cd ' + pipes.quote(os.path.join(buildspace, package.name)) + ' && '
                 # Then reproduce the command called
-                cmd += ' '.join(e.cmd) if isinstance(e.cmd, list) else e.cmd
+                if isinstance(e.cmd, list):
+                    # Escape individual arguments if possible
+                    for arg in e.cmd:
+                        cmd += ' ' + pipes.quote(arg)
```

Comment:
This will result in two consecutive spaces. Please use the join semantic to avoid that:

    cmd += ' '.join([quote(arg) for arg in e.cmd])